sympy/simplify/combsimp.py

```diff
@@ -66,9 +64,12 @@ def combsimp(expr):
 
     # as a rule of thumb, if the expression contained gammas initially, it
     # probably makes sense to retain them
-    as_gamma = expr.has(gamma)
     as_factorial = expr.has(factorial)
     as_binomial = expr.has(binomial)
+    as_gamma = expr.has(gamma)
+
+    if as_gamma and (not expr.args[0].is_Integer):
```

Comment:
`expr.args[0]` is not the argument of the `expr.func` which is not `gamma`, in general. For example,
```
>>> expr = gamma(n + S(1)/3)*gamma(n + S(2)/3)
>>> expr.func
<class 'sympy.core.mul.Mul'>
>>> expr.args[0]
gamma(n + 1/3)
>>> expr.args[1]
gamma(n + 2/3)
```
To find the arguments it is necessary to examine each application of `gamma` more closely.

But that is not necessary to do now. It can be done as a final touch. At this point the arguments need not be inspected.